New/Edit Contact: remove the Cancel button

mlutfy opened the pull request:
Overview
----------------------------------------

When creating a new Contact, or editing a contact, do not display a Cancel button, as it is unlikely to be used for its intended purpose.

Before
----------------------------------------

![image](https://user-images.githubusercontent.com/254741/178829900-6f443137-954e-4ac3-adc7-a20e1fab6ec8.png)

After
----------------------------------------

![image](https://user-images.githubusercontent.com/254741/178830005-7e60cfa3-4ecc-4b45-bb37-4b8535d0f25a.png)

Comments
----------------------------------------

- Using the Cancel button is rather unlikely. Users are more likely to hit back or use the navigation menu.
- If they do hit the Cancel button, it's probably by accident, which can be pretty frustrating.

mlutfy committed:
New Contact/Case: remove the Cancel button

mattwire commented:
Fine by me. Does this work ok without cancel button on popup forms?

yashodha commented:
@mlutfy 
Does this have a related ticket for this? Are there more screens that need to be consistent with this behavior? There is cancel button on almost all the forms there are.

mlutfy commented:
I incorrectly mentioned that this impacts only new Contacts, but it also impacts "Edit Contact". I still think it's fine, because clicking cancel after doing an edit would still be cringe. Updated description.

@mattwire re:popups, I guess New Contact is very rarely used in a popup, but New Case is (and is most often used in popups). So I reverted the change on New Case.

![image](https://user-images.githubusercontent.com/254741/179068042-90ab993b-8f1d-4c18-818f-75543bddf1af.png)

Technically, however, it does not make the popup bug if the cancel button is absent (I tested).

@yashodha I don't have a Gitlab issue. So far I removed Cancel from here: #23696, but that was an obvious candidate. I did grep for other occurrences of cancel, but I prefer to only PR when I see a clear reason to remove it. In admin screens, for example, it can be more debatable.

I get a lot of user feedback about the 1) excessive number of buttons 2) how easy it is to click the wrong one, 3) incorrect labels (often made worse by confused translators). However, I understand that UI changes can be very disruptive (see also #23831)

mlutfy commented:
I'm fine with closing this. I was thinking of maybe creating an extension "unclutter" or something that could be opt-in (https://chat.civicrm.org/civicrm/pl/oi9aehbi77n7jm46gjo999ajgc)
